Exercise/phone numbers

sushi-buffer opened the pull request:
Changes the input and expectations for tests to conform to the [canonical_data.json](https://github.com/exercism/problem-specifications/blob/main/exercises/phone-number/canonical-data.json)

kotp commented on exercises/practice/phone-number/test/phone_number_test.cljs:
```diff
@@ -27,7 +27,7 @@
+  (is (= "223" (phone-number/area-code "12234567890"))))
 
 (deftest pretty-print
   (is (= "(123) 456-7890" (phone-number/pretty-print "1234567890"))))
```
This one is incorrect as far as being invalid, so should not pretty print as it is.  (It should not get that far.)

kotp replied:
Thank you, I appreciate it!

kotp replied:
The track decides how to handle the error for the specific exercise (not sure what was decided here, but I would take the current test hints, as it was reviewed and brought in at some point) and I believe some tracks chose to raise an exception, while others chose to use all zeros.

Looking for confirmation from the @exercism/clojurescript team.

kotp replied:
Thank you!  Now to wait for someone else to review.

kotp reviewed (changes requested):
Double check the rest of them.

The names of the tests should match closely.  I do not notice, for example `invalid-if-area-code-starts-with-0-on-valid-11-digit-number` (if numbers are allowed in the name, even, for this language, and the test framework).

Is it worthwhile to add the rest of the tests in here (since we are here)?  This has about 9 tests, while there are about 20 tests in the canonical listing.

sushi-buffer committed:
Fix faulty test expectations and add missing tests.

Fixes tests with erroneous expectations for the 10th and 7th digits in the
number, as only numbers from 2 through 9 should be accepted. Adds some
missing tests from the canonical file.

LeaveNhA reviewed (approved):
Looks good to me! Thank you for your contribution @4lph4-Ph4un. 🙏🏻😇
